menu-animation/app.js

```diff
@@ -0,0 +1,19 @@
+import './style.css';
+import './menu.css';
+
+const ITEM_CLASS = 'menu-item';
+const ACTIVE_CLASS = 'active';
+
+document.querySelectorAll(`.${ITEM_CLASS}`).forEach(el => {
+    el.addEventListener('click', e => {
+        const target = e.currentTarget;
+        const prevActive = document.querySelector(`.${ITEM_CLASS}.${ACTIVE_CLASS}`);
+        if (prevActive && target !== prevActive) {
+            prevActive.classList.remove(ACTIVE_CLASS);
+        }
+
+        requestAnimationFrame(() => {
+            target.classList.add(ACTIVE_CLASS);
```

Comment:
Any reason to add the class inside requestAnimationFrame but didn't do it to remove? :)

One tip: instead of getting `prevActive` at each new click, you could instead set a global variable "currentActive" and update its value as the last thing on the click event. That way you don't need to perform querySelector at each click.